test/EFCore.PG.Tests/ApiConsistencyTestBase.cs

```diff
@@ -0,0 +1,949 @@
+        }
+
+        private string ValidateMutableMethod(MethodInfo mutableMethod)
+        {
+            var parameters = mutableMethod.GetParameters();
+            var parameterIndex = mutableMethod.IsStatic ? 1 : 0;
+            var firstParameter = parameters.Length > parameterIndex ? parameters[parameterIndex] : null;
+            var name = mutableMethod.Name;
+            if (firstParameter != null
+                && (name.StartsWith("Add", StringComparison.Ordinal)
+                    || name.StartsWith("Remove", StringComparison.Ordinal)
+                    || name.StartsWith("Set", StringComparison.Ordinal))
+                && !Fixture.MetadataMethodExceptions.Contains(mutableMethod)
+                && mutableMethod.ReturnType != firstParameter.ParameterType
+                && (firstParameter.ParameterType != typeof(Type) || mutableMethod.ReturnType != typeof(string))
+                && (firstParameter.ParameterType != typeof(string) || mutableMethod.ReturnType != typeof(FieldInfo))
+                && !Fixture.MutableMetadataTypes.ContainsKey(mutableMethod.ReturnType))
+            {
+                if (name.StartsWith("Set", StringComparison.Ordinal))
+                {
+                    if (mutableMethod.ReturnType != typeof(void))
+                    {
+                        return $"{mutableMethod.DeclaringType.Name}.{name}({Format(parameters)}) expected to have a void return type";
+                    }
+                }
+                else
+                {
+                    return $"{mutableMethod.DeclaringType.Name}.{name}({Format(parameters)}) expected to have an IMutable or " +
+                        $"{firstParameter.ParameterType.ShortDisplayName()} return type";
+                }
+            }
+
+            return null;
+        }
+
+        [ConditionalFact]
+        public void Convention_metadata_methods_have_expected_shape()
+        {
+            var errors =
+                Fixture.MetadataMethods
+                .SelectMany(m => m.Convention.Select(ValidateConventionMethod))
+                .Where(e => e != null)
+                .ToList();
+
+            Assert.False(
+                errors.Count > 0,
+                "\r\n-- Mismatches: --\r\n" + string.Join(Environment.NewLine, errors));
+        }
+
+        private string ValidateConventionMethod(MethodInfo conventionMethod)
+        {
+            var parameters = conventionMethod.GetParameters();
+            var parameterIndex = conventionMethod.IsStatic ? 1 : 0;
+            var firstParameter = parameters.Length > parameterIndex ? parameters[parameterIndex] : null;
+            var name = conventionMethod.Name;
+            if (firstParameter != null
+                && (name.StartsWith("Add", StringComparison.Ordinal)
+                    || name.StartsWith("Remove", StringComparison.Ordinal)
+                    || name.StartsWith("Set", StringComparison.Ordinal))
+                // TODO: "TEMPORARY" HACK
```

Comment:
@AndriySvyryd this is what we discussed

Reply:
You can add it to Fixture.MetadataMethodExceptions instead